src/test/java/com/kodilla/ecommercee/repositoryDao/CartRepositoryTestSuit.java

```diff
@@ -0,0 +1,224 @@
+    }
+    @Transactional
+    @Test
+    public void testCartDaoSaveAndFindAll(){
+        //Given
+        userDao.save(user1);
+        userDao.save(user2);
+        userDao.save(user3);
+        cartRepository.save(cart1);
+        cartRepository.save(cart2);
+        cartRepository.save(cart3);
+        //When
+        List<Cart> listOfCarts = cartRepository.findAll();
+        //Then
+        Assert.assertEquals(3,listOfCarts.size());
+        Assert.assertNotEquals(null,cartRepository.findById(cart1.getId()));
+        Assert.assertNotEquals(null,cartRepository.findById(cart2.getId()));
+        Assert.assertNotEquals(null,cartRepository.findById(cart3.getId()));
+        //Clean Up
+        try{
+            userDao.deleteById(user1.getId());
+            userDao.deleteById(user2.getId());
+            userDao.deleteById(user3.getId());
+            cartRepository.deleteById(cart1.getId());
+            cartRepository.deleteById(cart2.getId());
+            cartRepository.deleteById(cart3.getId());
+
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+    @Transactional
+    @Test
+    public void testCartDaoSaveAndFindById(){
+        //Given
+        cartRepository.save(cart1);
+        cartRepository.save(cart2);
+        cartRepository.save(cart3);
+        Long IdCart1 = cart1.getId();
+        //When
+        Optional<Cart> cartFound = cartRepository.findById(IdCart1);
+        //Then
+        Assert.assertEquals(cart1.getId(),cartFound.get().getId());
+        //Clean Up
+        try{
+            cartRepository.deleteById(cart1.getId());
+            cartRepository.deleteById(cart2.getId());
+            cartRepository.deleteById(cart3.getId());
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+    @Transactional
+    @Test
+    public void testCartDaoDeleteById(){
+        //Given
+        userDao.save(user1);
+        userDao.save(user2);
+        userDao.save(user3);
+        cartRepository.save(cart1);
+        cartRepository.save(cart2);
+        cartRepository.save(cart3);
+        Long cart2Id = cart2.getId();
+        //When
+        cartRepository.deleteById(cart2Id);
+        List<Cart> cartList = cartRepository.findAll();
+        //Then
+        Assert.assertFalse(cartList.contains(cart2));
+        //Clean Up
+        try{
+            cartRepository.deleteById(cart1.getId());
+            cartRepository.deleteById(cart3.getId());
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+
+    @Transactional
+    @Test
+    public void testCartOneOnOneWithOrder(){
+        //Given
+        Order order = new Order();
+        cart1.setOrder(order);
+        //When
+        cartRepository.save(cart1);
+        Assert.assertNotEquals(0,cartRepository.findById(cart1.getOrder().getOrderId()));
```

Comment:
drugi wystarczy zapisać jako order.getId()

Reply:
tak Ola, tylko wtedy odnioslbym sie bezposrednio do utworzonego obiektu a wydawalo mi sie ze chodzi o to w tym tescie by sprawdzic relacje Order-Cart z poziomu repozytorium dlatego wywolalem orderId w ten sposob

Reply:
by test udowodnil ze ta relacja istnieje